Author: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>

See issue #272 for more information. In short, we need to use EGL on Tizen
Mobile devices and OSMesa in the emulator (the EGL implementation in 2.1 does
not work as expected in the emulator and crashes Crosswalk).

The easiest way for this to work is to look for the GL implementations
available in a different order: if we ship libosmesa.so in the emulator RPM
package, we can look for OSMesa first and, if it is not available, we fall back
to EGL (which means a real device without libosmesa.so in the system) and only
then look for the Desktop GL implementation (Tizen PC, maybe).

Once we move to a Tizen version whose emulator image works correctly with EGL,
we can drop this patch.

Not upstreamable.
--- src/ui/gl/gl_implementation_x11.cc
+++ src/ui/gl/gl_implementation_x11.cc
@@ -52,9 +52,9 @@ base::NativeLibrary LoadLibrary(const char* filename) {
 }  // namespace
 
 void GetAllowedGLImplementations(std::vector<GLImplementation>* impls) {
-  impls->push_back(kGLImplementationDesktopGL);
-  impls->push_back(kGLImplementationEGLGLES2);
   impls->push_back(kGLImplementationOSMesaGL);
+  impls->push_back(kGLImplementationEGLGLES2);
+  impls->push_back(kGLImplementationDesktopGL);
 }
 
 bool InitializeGLBindings(GLImplementation implementation) {
